types/react-native-auth0/index.d.ts

```diff
@@ -75,15 +76,18 @@ export interface ResetPasswordParams {
     connection: string;
 }
 
-export interface UserInfo {
+export type UserInfo = {
     email: string;
     emailVerified: boolean;
     name: string;
     nickname: string;
     picture: string;
     sub: string;
     updatedAt: string;
-}
+} & {
+    [key: string]: string;
```

Comment:
Could maybe make more sense to generalize `UserInfo` to accept a type of the custom claims? That would make possible to define a meaningful type and export it around the app

Reply:
I don't really know how that would work. The claims are part of the actual properties of the UserInfo, not contained within a single property, eg.
`{
  "email": "foo@bar.org",
  "emailVerified": true,
  "https://acme.com/foo": "bar"
}`

The only thing that differentiates them from the other properties, is that the key should always be a valid url format, using http or https protocol.

Normally I would just have added the `[key: string]: string;` next to the other properties of the UserInfo interface, but this is not allowed due to not all properties being of type string.

Maybe this isn't an issue, and I am wrong with limiting the values of the claims to strings, but that is the only thing I tried until now.

Reply:
In this case, I think generics address your problem:
you can define the `UserInfo` to accept a *type parameter*, i.e. `type UserInfo<Claims = {}> = {email: string, emailVerified: boolean} & Claims`.
In this way you can extend the base `UserInfo` with your *claims*.

Does it make sense?

Can you give it a try?

Reply:
I would go for the generic implementation as the *type parameter* is defaulted to `{}` so is an opt-in for the users who want more type safety, while it stays simple for those who need just the plain `UserInfo`.

The comment could be helpful, but I think if deductible from the type name - being in the `react-native-auth0` context